MapboxDirections/MBDirections.swift

```diff
@@ -229,11 +245,26 @@ open class Directions: NSObject {
      */
     @objc(URLForCalculatingDirectionsWithOptions:)
     open func url(forCalculating options: RouteOptions) -> URL {
-        let params = options.params + [
-            URLQueryItem(name: "access_token", value: accessToken),
-        ]
-        
-        let unparameterizedURL = URL(string: options.path, relativeTo: apiEndpoint)!
+        let isMapboxHost = apiEndpoint.host == mapboxHost
+        var params = options.params
+        if isMapboxHost {
+            params += [
+                URLQueryItem(name: "access_token", value: accessToken),
+            ]
+        }
+
+        let path: String
+        if isMapboxHost {
+            path = options.path
+        }
+        else
+        {
+            assert(!options.queries.isEmpty, "No query")
+            let queryComponent = options.queries.joined(separator: ";")
+            path = "route/v1/\(options.profileIdentifier.rawValue)/\(queryComponent)"
+        }
+
+        let unparameterizedURL = URL(string: path, relativeTo: apiEndpoint)!
```

Comment:
This should be properly unwrapped.

Reply:
I did not changed this unparameterizedURL unwrap... It is the original code.